### PR TITLE
Move to mainline Microkit

### PR DESCRIPTION
**Pull request by Ivan-Velickovic**
Moves sDDF to use 1.4.0 of Microkit which is planned to be released later today. CI is expected to fail until the release happens.

Merge checklist:
* [x] Update README
* [x] Update CI to grab Microkit 1.4.0

### CODE REVIEW

**Comment by Ivan-Velickovic** [comment]
Not dealing with the style check in this PR.

File: drivers/network/virtio/ethernet.c [hunk 1]
```diff
@@ -456,7 +456,7 @@ void notified(microkit_channel ch)
     switch (ch) {
     case IRQ_CH:
         handle_irq();
-        microkit_irq_ack_delayed(ch);
+        microkit_irq_ack(ch);
```
Comment (Ivan-Velickovic) [inline drivers/network/virtio/ethernet.c:459]: This should be a deferred IRQ ack instead

File: examples/serial/mqlog
Comment (Ivan-Velickovic) [inline examples/serial/mqlog:1]: this should be removed

File: examples/timer/mqlog
Comment (Ivan-Velickovic) [inline examples/timer/mqlog:1]: same here… whoops

**Review by alexandermbrown — DISMISSED** [review]
LGTM. I built `echo_server` on macOS as a sanity check and it seems good.

**Comment by alexandermbrown** [comment]
I don't think this is related to this PR but `gen_config.h` in the microkit sdk still uses `CONFIG_PLAT_QEMU_ARM_VIRT`. Is this ok?

**Comment by Ivan-Velickovic** [comment]
> I don't think this is related to this PR but gen_config.h in the microkit sdk still uses CONFIG_PLAT_QEMU_ARM_VIRT. Is this ok?

That's fine.

**Review by alexandermbrown — APPROVED** [review]

**Pull request merged**